### PR TITLE
Switch to new feed without BlogMonster

### LINKED ISSUE #230: BlogMonster is slow 🐌
Building the feed after having checked for broken feeds takes between 2 and 3 minutes.

It is freakishly slow as fetching the feeds themselves takes less than 15 seconds when done correctly. To me it seems like BlogMonster is not running parallel inside of its belly.

### PR DESCRIPTION
**Pull request by Cheesebaron**
We have a hacked together version of combined feed currently, which primarily uses blog monster to combine the downloaded feeds and for some caching of feeds for an hour.

This PR decouples us from BlogMonster but aslo adds a simple retry mechanism using a Polly policy. Also instead of the caching mechanism BlogMonster provides, we are now using Polly to help us with that, which simplifies the code significantly.
Also it seems like the BlogMonster project is pretty much dead, so it seems fit to do something about that.

TODO:
- [x] Fix Test suite

Fixes #230
Helps us getting along with #536 and #588 when we switch to ASP.NET Core

### CODE REVIEW

**Comment by Cheesebaron** [comment]
@planetxamarin/owners can I get a couple of reviews on this one? I am very confident it works. However, would love some eyes on this.

In general, the feed loads in around 15 or 16 seconds now if all feeds respond in a timely manner. A huge improvement over our current Blog Monster based feed, which can take up to a minute.

**Comment by nickrandolph** [comment]
I'll try to take a look over the weekend. Do you know how much effort it would be to move to new project system as well?

**Comment by Cheesebaron** [comment]
@icebeam7 where is it hosted, it often takes more than 10-15 seconds to reach it? This is around the amount of time it normally takes to fetch _all_ the 100+ feeds we have!

**Comment by jfversluis** [comment]
@icebeam7 I see the same thing. Over 20 seconds even to load your regular website.

image

File: src/Firehose.Web/Infrastructure/NewCombinedFeedSource.cs [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+        private void EnsureHttpClient()
+        {
+            if (httpClient == null)
+            {
+                httpClient = new HttpClient();
+                httpClient.DefaultRequestHeaders.UserAgent.Add(
+                    new ProductInfoHeaderValue("PlanetXamarin", $"{GetType().Assembly.GetName().Version}"));
+                httpClient.Timeout = TimeSpan.FromSeconds(10);
+
+                ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12 | SecurityProtocolType.Tls11;
```
Comment (nickrandolph) [inline src/Firehose.Web/Infrastructure/NewCombinedFeedSource.cs]: should this be changed to remove tls11 and add tls13?

Comment (Cheesebaron) [reply src/Firehose.Web/Infrastructure/NewCombinedFeedSource.cs]: Yeah, I guess that would be the better practice.

Comment (Cheesebaron) [reply src/Firehose.Web/Infrastructure/NewCombinedFeedSource.cs]: TSL13 doesn't seem to be available in full framework

File: src/Firehose.Web/Firehose.Web.csproj [hunk 2]
```diff
@@ -196,7 +196,6 @@
     <Compile Include="App_Start\RouteConfig.cs" />
     <Compile Include="Authors\AllanRitchie.cs" />
     <Compile Include="Authors\AndrewHoefling.cs" />
-    <Compile Include="Authors\BenceLenart.cs" />
```
Comment (nickrandolph) [inline src/Firehose.Web/Firehose.Web.csproj]: I just tried the feed and it seems to return data. I did however notice that the uri for the website url is missing the "h" in http.

File: src/Firehose.Web/Authors/NigelSampson.cs [hunk 3]
```diff
@@ -1,33 +0,0 @@
...
-		public string FirstName => "Nigel";
-		public string LastName => "Sampson";
-		public string StateOrRegion => "New Zealand";
-		public string EmailAddress => "nigel.sampson@compiledexperience.com";
-		public string ShortBioOrTagLine => "is the project lead of Caliburn.Micro and is often a developer at Pushpay";
-		public Uri WebSite => new Uri("https://compiledexperience.com/");
-		public string TwitterHandle => "nigelsampson";
-		public string GitHubHandle => "nigel-sampson";
-		public string GravatarHash => "ffccec4c43820d5370ad22de8d569d77";
-		public IEnumerable<Uri> FeedUris => new [] { new Uri("https://compiledexperience.com/blog/syndicate/atom/") };
```
Comment (nickrandolph) [inline src/Firehose.Web/Authors/NigelSampson.cs]: If we change the feed to https://compiledexperience.com/blog/syndicate/rss/ it should parse.
I suspect we may still want to exclude based on lack of xamarin content

File: src/Firehose.Web/Firehose.Web.csproj (hunk 2 above)
Comment (Cheesebaron) [reply src/Firehose.Web/Firehose.Web.csproj]: I guess he has fixed it. When I fetched the feed during my test it didn't contain any items.

**Comment by Cheesebaron** [comment]
@planetxamarin/owners I've rebased the PR and removed 3 authors:

- @nigel-sampson does not have any Xamarin related posts in feed
- @achrafbenalaya does not provide proper Atom or RSS feed
- @icebeam7 does not meet performance requirements (currently set at 15 seconds timeout) most of the time it exceeds that, unless the feed is hot

Without these authors, getting all author feeds takes _8 seconds_!

**Comment by Cheesebaron** [comment]
@jamesmontemagno how do we get this into a staging environment?

**Comment by jamesmontemagno** [comment]
I created a branch, let's push it onto there and i can get it going

**Comment by Cheesebaron** [comment]
@jamesmontemagno don't think I want to update to netcore in this PR, it is not just a single line change

**Comment by jamesmontemagno** [comment]
ah, gotcha, let's just create a new branch and then i can test it out there. i thought this was something else lol.

**Comment by Cheesebaron** [comment]
The branch is on this repo now too.

**Comment by jamesmontemagno** [comment]
do you just want to push all the code there and then open a pr
and i'll get it building

**Pull request closed without merging**